File: Leetcode/python/2785.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = "AEIOUaeiou"
        vowel_list = []

        for c in s:
            if c in vowels:
                vowel_list.append(c)
            
        vowel_list.sort()

        result = []
        i = 0
        for c in s:
            if c in vowels:
                result.append(vowel_list[i])
                i += 1
            else:
                result.append(c)
        
        return "".join(result)

File: Leetcode/2785.cs
using System.Text;

namespace Leetcode
{
    internal class _2785
    {
        public string SortVowels(string s)
        {
            Span<bool> isVowel = stackalloc bool[128];
            foreach (char c in "aeiouAEIOU")
            {
                isVowel[c] = true;
            }

            var vowels = new List<char>();
            foreach (char c in s)
            {
                if (isVowel[c])
                {
                    vowels.Add(c);
                }
            }

            vowels.Sort();

            var res = new StringBuilder(s.Length);
            int j = 0;
            foreach (char c in s)
            {
                if (isVowel[c])
                {
                    res.Append(vowels[j++]);
                }
                else
                {
                    res.Append(c);
                }
            }

            return res.ToString();
        }
    }
}
